File: csharp/resistor-color/ResistorColor.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ResistorColor
{
    private static readonly string[] colors = new[]{ "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };

    public static int ColorCode(string color) => Array.IndexOf(colors, color);

    public static string[] Colors() => colors;
}

File: java/resistor-color/src/main/java/ResistorColor.java
class ResistorColor {

    private final static String[] COLORS = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};

    int colorCode(String color) {
        for(int i=0; i < COLORS.length; ++i){
            if(color.equals(COLORS[i]))
                return i;
        }

        throw new IllegalArgumentException();
    }

    String[] colors() {
        return COLORS;
    }
}
